client/fingerprint/vault.go

```diff
@@ -20,73 +22,121 @@ const (
-	if f.client == nil {
-		vaultConfig, err := config.VaultConfig.ApiConfig()
+	if state.client == nil {
+		vaultConfig, err := cfg.ApiConfig()
 		if err != nil {
-			return fmt.Errorf("Failed to initialize the Vault client config: %v", err)
+			return fmt.Errorf("Failed to initialize the Vault client config for %s: %v", cfg.Name, err)
 		}
-		f.client, err = vapi.NewClient(vaultConfig)
+		state.client, err = vapi.NewClient(vaultConfig)
 		if err != nil {
-			return fmt.Errorf("Failed to initialize Vault client: %s", err)
+			return fmt.Errorf("Failed to initialize Vault client for %s: %s", cfg.Name, err)
 		}
-		useragent.SetHeaders(f.client)
+		useragent.SetHeaders(state.client)
 	}
 
 	// Connect to vault and parse its information
-	status, err := f.client.Sys().SealStatus()
+	status, err := state.client.Sys().SealStatus()
 	if err != nil {
-
 		// Print a message indicating that Vault is not available anymore
-		if f.lastState == vaultAvailable {
-			f.logger.Info("Vault is unavailable")
+		if state.isAvailable {
+			logger.Info("Vault is unavailable", "cluster", cfg.Name)
 		}
-		f.lastState = vaultUnavailable
+		state.isAvailable = false
+		state.nextCheck = time.Time{} // always check on next interval
 		return nil
 	}
 
-	resp.AddAttribute("vault.accessible", strconv.FormatBool(true))
-	// We strip the Vault prefix because < 0.6.2 the version looks like:
-	// status.Version = "Vault v0.6.1"
-	resp.AddAttribute("vault.version", strings.TrimPrefix(status.Version, "Vault "))
-	resp.AddAttribute("vault.cluster_id", status.ClusterID)
-	resp.AddAttribute("vault.cluster_name", status.ClusterName)
+	if cfg.Name == "default" {
+		resp.AddAttribute("vault.accessible", strconv.FormatBool(true))
+		resp.AddAttribute("vault.version", strings.TrimPrefix(status.Version, "Vault "))
+		resp.AddAttribute("vault.cluster_id", status.ClusterID)
+		resp.AddAttribute("vault.cluster_name", status.ClusterName)
+	} else {
+		resp.AddAttribute(fmt.Sprintf("vault.%s.accessible", cfg.Name), strconv.FormatBool(true))
+		resp.AddAttribute(fmt.Sprintf("vault.%s.version", cfg.Name), strings.TrimPrefix(status.Version, "Vault "))
+		resp.AddAttribute(fmt.Sprintf("vault.%s.cluster_id", cfg.Name), status.ClusterID)
+		resp.AddAttribute(fmt.Sprintf("vault.%s.cluster_name", cfg.Name), status.ClusterName)
+	}
 
 	// If Vault was previously unavailable print a message to indicate the Agent
 	// is available now
-	if f.lastState == vaultUnavailable {
-		f.logger.Info("Vault is available")
+	if !state.isAvailable {
+		logger.Info("Vault is available", "cluster", cfg.Name)
 	}
-	f.lastState = vaultAvailable
+
+	// Widen the minimum window to the next check so that if one out of a set of
+	// Vaults is unhealthy we don't greatly increase requests to the healthy
+	// ones. This is less than the minimum window if all Vaults are healthy so
+	// that we don't desync from the larger window provided by Periodic
+	state.nextCheck = time.Now().Add(30 * time.Second)
```

Comment:
```suggestion
	// Widen the minimum window to the next check so that if one out of a set of
	// Vaults is unhealthy we don't greatly increase requests to the healthy
	// ones. This is less than the minimum window if all Vaults are healthy so
	// that we don't desync from the larger window provided by Periodic
	state.nextCheck = time.Now().Add(29 * time.Second)
```
`RandomStargger` may return `0`, which would cause `Periodic` and `nextCheck` to be both `30s`. I think that, depending on the timing, it may result in the a healthy cluster being skipped?